### ISSUE TITLE
Checksum verification not available through API, making --force necessary

### ISSUE THREAD

**Issue by yingw787** [Kaggle/kaggle-api#233]
Hello!

I was doing a coding challenge and noticed that the Kaggle API did not yet support checksum verification. You could name any file dataset.zip, and if Kaggle detected that the file with the same name existed, it would silently exit and say the file was successfully downloaded. I was wondering whether there are plans on the roadmap to include checksum verification of downloaded files, and whether Kaggle generates and caches checksums for datasets at upload time. I think this is important in order to effectively make the API scriptable and to effectively include Kaggle as part of some workflow pipelines. As a workaround, I am generating checksums myself, matching them with known datasets, and including them within my method calls on top of the API to know when to force a re-download. Otherwise I'm not sure if Kaggle already has this feature and I'm just using the API wrong.

If this is a feature gap and Kaggle is open for open source contributors, please let me know.

Thanks
Ying

**Comment by neomatrix369** [comment]
Yes this is a handy feature, at the moment I can't sync datasets between a local machine and Kaggle account without having to upload/download the whole dataset, this is inefficient and slow.
